File: tests/human/x/go/dataset_where_filter.go
//go:build ignore

package main

import "fmt"

type Person struct {
	Name string
	Age  int
}

func main() {
	people := []Person{
		{"Alice", 30},
		{"Bob", 15},
		{"Charlie", 65},
		{"Diana", 45},
	}

	var adults []struct {
		Name     string
		Age      int
		IsSenior bool
	}

	for _, p := range people {
		if p.Age >= 18 {
			adults = append(adults, struct {
				Name     string
				Age      int
				IsSenior bool
			}{p.Name, p.Age, p.Age >= 60})
		}
	}

	fmt.Println("--- Adults ---")
	for _, person := range adults {
		note := ""
		if person.IsSenior {
			note = " (senior)"
		}
		fmt.Printf("%s is %d%s\n", person.Name, person.Age, note)
	}
}

File: tests/human/x/java/DatasetWhereFilter.java
import java.util.*;

class Person {
    String name;
    int age;
    boolean isSenior;
    Person(String name, int age) { this.name = name; this.age = age; this.isSenior = age >= 60; }
}

public class DatasetWhereFilter {
    public static void main(String[] args) {
        List<Person> people = Arrays.asList(
            new Person("Alice", 30),
            new Person("Bob", 15),
            new Person("Charlie", 65),
            new Person("Diana", 45)
        );

        List<Person> adults = new ArrayList<>();
        for (Person p : people) {
            if (p.age >= 18) {
                adults.add(new Person(p.name, p.age));
            }
        }

        System.out.println("--- Adults ---");
        for (Person p : adults) {
            String extra = p.isSenior ? " (senior)" : "";
            System.out.println(p.name + " is " + p.age + extra);
        }
    }
}
